src/models/Scholarships.ts

```diff
@@ -70,7 +70,9 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
     }
     // Sort ties by deadline earliest to earliest
     if (opts.sortField !== 'deadline') {
-      query = query.orderBy('deadline', 'asc');
+      query = query
+        .where('deadline', '>', new Date())
```

Comment:
3 issues with this:
1. This is inflexible. The caller has no option to show expired scholarships, which would be relevant in other use cases.
2. This is only executed if you're not sorting by deadline.
3. You have no tests to prove the correctness of this. You should add test cases to verify this is correct.

Reply:
One more issue based on the failed tests but isn't obvious: Firestore doesn't let you actually do this unless the sorting field is `deadline` so you'll need to handle both cases.